File: Array/stack.c
#include <stdio.h>
#define MAX_SIZE 10
int arr[MAX_SIZE];
int size;
int isFull = 0, isEmpty = MAX_SIZE;
void checkStackSpace();
void inputStack()
{
    for (int i = 0; i < size; i++)
    {
        printf("Enter arr[%d] element: ", i);
        scanf("%d", &arr[i]);
    }
    isFull = size;
    isEmpty = 100 - size;
}

void traverseStack()
{
    for (int i = 0; i < size; i++)
    {
        printf("| %d |", arr[i]);
    }
    checkStackSpace();
}

void pushStack()
{
    if (isFull == MAX_SIZE)
    {
        printf("Stack is full! Overflow condition occurs.\n");
    }
    else
    {
        int value;
        printf("Enter value: ");
        scanf("%d", &value);
        arr[size] = value;
        size++;
        isFull++;
        isEmpty--;
        printf("Pushed element are: %d\n", value);
        printf("Successfully pushed!");
        checkStackSpace();
    }
}

void popStack()
{
    if (isFull == 0)
    {
        printf("Stack is empty! underflow condition occurs..\n");
    }
    else
    {
        int popElement = arr[size - 1];
        size--;
        isEmpty++;
        isFull--;
        printf("Pop element are: %d\n", popElement);
        printf("Successfully popped!");
        checkStackSpace();
    }
}

void sortStack()
{
    for (int i = 0; i < size - 1; i++)
    {
        for (int j = 0; j < size - i - 1; j++)
        {

            if (arr[j] > arr[j + 1])
            {
                int tmp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = tmp;
            }
        }
    }
    printf("Stack sorted successfully\n");
}

void stackPeek()
{
    if (isFull == 0)
        printf("The stack is empty!...");
    else
        printf("Top element of the stack is: %d", arr[size - 1]);
}
void checkStackSpace()
{
    printf("\nStack Full area: %d", isFull);
    printf("\nStack Empty area: %d", isEmpty);
}
void stackCount()
{
    printf("Total element in the stack is: %d\n", isFull);
}
int main()
{
    // printf("Enter the size of an array: ");
    // scanf("%d", &size);
    // if (size <= 0 || size > MAX_SIZE)
    // {
    //     printf("Invalid array size!");
    // }
    // else
    // {
    //     inputStack();
    // }

    int choice;
    printf("\n\n");
    while (1)
    {
        printf("\n1. for Display Stack");
        printf("\n2. for Push");
        printf("\n3. for Pop");
        printf("\n4. for Sort");
        printf("\n5. for Peek");
        printf("\n6. for Check Stack Space");
        printf("\n7. for Check Stack Count");
        printf("\n8. for Exit");
        printf("\nEnter your choice: ");
        scanf("%d", &choice);

        if (choice == 8)
            break;
        else
        {

            switch (choice)
            {
            case 1:
                traverseStack();
                break;
            case 2:
                pushStack();
                break;
            case 3:
                popStack();
                break;
            case 4:
                sortStack();
                break;
            case 5:
                stackPeek();
                break;
            case 6:
                checkStackSpace();
                break;
            case 7:
                stackCount();
                break;
            default:
                printf("Enter a valid choice: ");
                break;
            }
        }
        printf("\n\n");
    }

    return 0;
}

File: stack.java
import java.util.Scanner;

public class stack {
    int MAX_SIZE = 10;
    int[] stack = new int[MAX_SIZE];
    int top = -1;
    int size;

    void displayStack() {
        if (top < 0)
            System.out.println("Stack is Empty!..");
        else {
            for (int i = top; i >= 0; i--) {
                System.out.print(" | ");
                System.out.print(stack[i]);
                System.out.print(" | ");
            }
        }
    }

    void checkSpace() {
        System.out.println("Stack total size: " + MAX_SIZE);
        System.out.println("Stack total size Full: " + (top + 1));
        System.out.println("Stack total size Empty: " + (MAX_SIZE - size));
    }

    void push() {
        if (top == MAX_SIZE - 1) {
            System.out.println("Stack is Full!...Overflow!!!!");
        } else {

            Scanner sc = new Scanner(System.in);
            System.out.print("Enter the element: ");
            int data = sc.nextInt();
            top = top + 1;
            size++;
            stack[top] = data;
            System.out.println(data + " is pushed....");
            checkSpace();
        }
    }

    void pop() {
        if (top < 0) {
            System.out.println("Stack is Empty....Underflow!!!!");
        } else {
            int data = stack[top];
            top = top - 1;
            size--;
            System.out.println(data + " is popped...");
            checkSpace();
        }
    }

    void count() {
        if (top < 0)
            System.out.println("Stack is Empty");
        else
            System.out.println("Count is: " + (top + 1));
    }

    void peek() {
        System.out.println("Top of the element is: " + stack[top]);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        stack obj = new stack();
        obj.checkSpace();
        while (true) {
            System.out.println("\n1. for Display Stack");
            System.out.println("\n2. for Push");
            System.out.println("\n3. for Pop");
            System.out.println("\n4. for Peek");
            System.out.println("\n5. for Check Stack Count");
            System.out.println("\n6. for Exit");
            System.out.print("Enter your choice: ");
            int choice = sc.nextInt();
            if (choice == 6)
                break;
            switch (choice) {
                case 1:
                    obj.displayStack();
                    break;
                case 2:
                    obj.push();
                    break;
                case 3:
                    obj.pop();
                    break;
                case 4:
                    obj.peek();
                    break;
                case 5:
                    obj.count();
                    break;
                default:
                    System.out.println("Invalid choice!..");
                    break;

            }
        }
        System.out.println("Thanks for using this APP....");
        System.out.println("Hope you are enjoy it!");

    }
}